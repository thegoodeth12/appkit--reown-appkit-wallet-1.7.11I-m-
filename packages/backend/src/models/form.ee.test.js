import { describe, it, expect } from 'vitest';
import User from '@/models/user';
import Base from '@/models/base';
import Form from '@/models/form.ee';

describe('Form model', () => {
  it('tableName should return correct name', () => {
    expect(Form.tableName).toBe('forms');
  });

  it('jsonSchema should have correct validations', () => {
    expect(Form.jsonSchema).toMatchSnapshot();
  });

  it('relationMappings should return correct associations', () => {
    const relationMappings = Form.relationMappings();

    const expectedRelations = {
      user: {
        relation: Base.BelongsToOneRelation,
        modelClass: User,
        join: {
          from: 'forms.user_id',
          to: 'users.id',
        },
      },
    };

    expect(relationMappings).toStrictEqual(expectedRelations);
  });
});
