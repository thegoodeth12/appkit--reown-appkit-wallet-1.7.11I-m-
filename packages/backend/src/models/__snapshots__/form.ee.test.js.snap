// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form model > jsonSchema should have correct validations 1`] = `
{
  "properties": {
    "createdAt": {
      "type": "string",
    },
    "description": {
      "type": "string",
    },
    "fields": {
      "type": "array",
    },
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "name": {
      "minLength": 1,
      "type": "string",
    },
    "response_message": {
      "type": "string",
    },
    "updatedAt": {
      "type": "string",
    },
  },
  "required": [
    "name",
  ],
  "type": "object",
}
`;
